### PR TITLE
.Net: PromptExecutionSettings in KernelArguments should be an ordered dictionary

### LINKED ISSUE #4212: .Net: PromptExecutionSettings in KernelArguments should be an ordered dictionary
tl;dr, it's not possible to pass multiple execution settings (one for each model ID, service ID, or another key) into an `IAIServiceSelector`

I have the following prompt

```csharp
var chat = kernel.CreateFunctionFromPrompt(
    new PromptTemplateConfig()
    {
        Name = "Chat",
        Description = "Chat with the assistant.",
        Template = @"{{ConversationSummaryPlugin.SummarizeConversation $history}}
        User: {{$request}}
        Assistant: ",
        TemplateFormat = "semantic-kernel",
        InputVariables = [
            new() { Name = "history", Description = "The history of the conversation.", IsRequired = false, Default = "" },
            new() { Name = "request", Description = "The user's request.", IsRequired = true }
        ],
        ExecutionSettings = [
            new() {
                // Default settings for all models
                ExtensionData = new() {
                    { "MaxTokens", 1000 },
                    { "Temperature", 0 }
                }
            },
            new() {
                // Settings for gpt-3.5-turbo
                ModelId = "gpt-3.5-turbo",
                ExtensionData = new() {
                    { "MaxTokens", 4000 },
                    { "Temperature", 0.1 }
                }
            },
            new() {
                // Settings for gpt-4
                ModelId = "gpt-4-1106-preview",
                ExtensionData = new() {
                    { "MaxTokens", 8000 },
                    { "Temperature", 0.3 }
                }
            }
        ]
    }
);
```

Notice that I have settings to support 3 different models. During invocation time, however, I want to override these settings using `KernelArguments`. Unfortunately, `KernelArguments` only lets me define a single `PromptExecutionSetting` and at this point, I don't know which model is best because that's the responsibility of the `IAIServiceSelector`

```csharp
var result = await kernel.InvokeAsync<string>(chat, new(new PromptExecutionSettings()
{
    ModelId = "gpt-3.5-turbo",
    ExtensionData = new() { { "MaxTokens", 1000 } }
}) { { "input", "Hello world!" } });
```

Once I'm in the `IAIServiceSelector`, I only get a single `PromptExecutionSettings`, so if the service selector chooses a different model than I provided in the `KernelArguments`, then I won't be able to override it corectly

```csharp
public bool TrySelectAIService<T>(
        Kernel kernel, KernelFunction function, KernelArguments arguments,
        [NotNullWhen(true)] out T? service, out PromptExecutionSettings? serviceSettings) where T : class, IAIService
```

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

Closes #4212 

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelArguments.cs [hunk 1]
```diff
@@ -40,7 +44,7 @@ public KernelArguments(PromptExecutionSettings? executionSettings = null)
     /// If <paramref name="executionSettings"/> is non-null, it is used as the <see cref="ExecutionSettings"/> for this new instance.
     /// Otherwise, if the source is a <see cref="KernelArguments"/>, its <see cref="ExecutionSettings"/> are used.
     /// </remarks>
-    public KernelArguments(IDictionary<string, object?> source, PromptExecutionSettings? executionSettings = null)
+    public KernelArguments(IDictionary<string, object?> source, Dictionary<string, PromptExecutionSettings>? executionSettings = null)
```
Comment (dmytrostruk) [inline dotnet/src/SemanticKernel.Abstractions/Functions/KernelArguments.cs:47]: Proposal: Maybe we should keep `KernelArguments` constructor that accepts `PromptExecutionSettings? executionSettings = null` for easier usability, if we want to pass just default settings. 

We can add new constructor that accepts `Dictionary<string, PromptExecutionSettings>? executionSettings` for more advanced scenarios.

**Review by dmytrostruk — APPROVED** [review]

File: dotnet/src/SemanticKernel.UnitTests/Functions/KernelArgumentsTests.cs [hunk 2]
```diff
@@ -113,6 +113,6 @@ public void ItCanBeInitializedFromAnotherSettingsInstance()
         Assert.True(sut.ContainsName("fake-key"));
         Assert.Equal("fake-value", sut["fake-key"]);
 
-        Assert.Same(executionSettings, sut.ExecutionSettings);
+        Assert.Same(executionSettings, sut.ExecutionSettings?["default"]);
```
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel.UnitTests/Functions/KernelArgumentsTests.cs:116]:
```suggestion
        Assert.Same(executionSettings, sut.ExecutionSettings?[PromptExecutionSettings.DefaultServiceId]);
```

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelArguments.cs (hunk 1 above)
Comment (markwallace-microsoft) [reply dotnet/src/SemanticKernel.Abstractions/Functions/KernelArguments.cs:47]: Yes, we have that on line 28

File: dotnet/src/SemanticKernel.UnitTests/Functions/KernelArgumentsTests.cs (hunk 2 above)
Comment (markwallace-microsoft) [reply dotnet/src/SemanticKernel.UnitTests/Functions/KernelArgumentsTests.cs:116]: Will make this change in the follow up PR

**Review by RogerBarreto — APPROVED** [review]

**Pull request merged**